### PR TITLE
feat: incremental updates for mobile transformer

### PR DESCRIPTION
**Pull request by pauldambra**
first pass at adding incremental mutations 

created some fake mobile recordings locally and these behaved as expected

this adds:

 * removals
     * these can be passed through - they're the same as rrweb
 * adds
     * the added wireframes can be converted just like in full snapshots (I think)
 * updates
     * these are implemented as removes and then adds
     * this might not work on recordings with very many mutations but it'll be at least enough for us to get some folk testing it and we _should_ have few enough mutations that it'll be ok

### CODE REVIEW

**Comment by marandaneto** [comment]
More like a question, since we have synthetic IDs, what if the removed/updated wireframe is a synthetic ID? Would that work the way how it is implemented? I believe the synthetic IDs are just the nested components right?

For example, a `button` on Mobile is a `div` and a nested `text`, if I remove or update the `button`, is the whole `div` and `text` correctly mutated?

**Review by marandaneto — APPROVED** [review]
Left a [comment](https://github.com/PostHog/posthog/pull/19567#issuecomment-1875030166) but otherwise LGTM

File: ee/frontend/mobile-replay/transformers.ts [hunk 1]
```diff
@@ -775,35 +770,76 @@ function chooseConverter<T extends wireframe>(
...
-        const convertedEl = converter(wireframe, children)
+        const convertedEl = convertWireframe(wireframe)
         if (convertedEl !== null) {
             acc.push(convertedEl)
         }
         return acc
     }, [] as serializedNodeWithId[])
 }
 
+function isMobileIncrementalSnapshotEvent(x: unknown): x is MobileIncrementalSnapshotEvent {
```
Comment (daibhin) [inline ee/frontend/mobile-replay/transformers.ts:793]: Feels like this does a little more than just checking the type on the event
```suggestion
function isValidMobileIncrementalSnapshotEvent(x: unknown): x is MobileIncrementalSnapshotEvent {
```

**Review by daibhin — APPROVED** [review]

File: ee/frontend/mobile-replay/transformers.ts (hunk 1 above)
Comment (pauldambra) [reply ee/frontend/mobile-replay/transformers.ts:793]: hmmm, I want to split this up more so I'm going to punt this into the future

**Pull request merged**